perllib/FixMyStreet/Cobrand/Bexley/Garden.pm

```diff
@@ -32,55 +34,127 @@ sub garden_service_ids {
     return [ 'GA-140', 'GA-240' ];
 }
 
+sub lookup_subscription_for_uprn {
+    my ($self, $uprn) = @_;
+
+    my $sub = {
+        row => undef,
+
+        email => undef,
+        cost => undef,
+        end_date => undef,
+        customer_external_ref => undef,
+        bins_count => undef,
+    };
+
+
+    my ( $customer, $contract );
+
+    my $results = $self->agile->CustomerSearch($uprn);
+
+    # find the first 'ACTIVATED' Customer with an 'ACTIVE'/'PRECONTRACT' contract
+    my $customers = $results->{Customers} || [];
+    OUTER: for ( @$customers ) {
+        next unless $_->{CustomertStatus} eq 'ACTIVATED'; # CustomertStatus (sic) options seem to be ACTIVATED/INACTIVE
+        my $contracts = $_->{ServiceContracts} || [];
+        next unless $contracts;
+        $customer = $_;
+        for ( @$contracts ) {
+            next unless $_->{ServiceContractStatus} =~ /(ACTIVE|PRECONTRACT)/; # Options seem to be ACTIVE/NOACTIVE/PRECONTRACT
+            $contract = $_;
+            # use the first matching customer/contract
+            last OUTER if $customer && $contract;
+        }
+    }
+
+    return unless $customer && $contract;
+
+    # XXX should maybe sort by CreatedDate rather than assuming first is OK
+    $sub->{cost} = try {
+        my ($payment) = grep { $_->{PaymentStatus} eq 'Paid' } @{ $contract->{Payments} };
+        return $payment->{Amount};
+    };
+
+    my $parser = DateTime::Format::Strptime->new( pattern => '%d/%m/%Y %H:%M' );
+    $sub->{end_date} = $parser->parse_datetime( $contract->{EndDate} );
+
+    $sub->{customer_external_ref} = $customer->{CustomerExternalReference};
+
+    $sub->{bins_count} = $contract->{WasteContainerQuantity};
+
+    my $c = $self->{c};
+    my $p = $c->model('DB::Problem')->search({
+        category => 'Garden Subscription',
```

Comment:
So looking at this orig_sub stuff again - I don't think we need to be fetching the DB data here (it doesn't look like $sub->{row} is used in garden_current_subscription below, is it used anywhere else?). I think we need to change the existing `get_original_sub` in Controller/Waste.pm, and rely on that to set `$c->stash->{orig_sub}`.

It's worth noting that `get_original_sub` a) does a check for title (so no cancellation reports are fetched) and b) does not do an external_id check. So having a separate lookup for Bexley here that is a bit different is possibly breaking assumptions further down the line 🤔